src/Uno.UI/UI/Xaml/Controls/SKCanvasElement.skia.cs

```diff
@@ -0,0 +1,108 @@
+		_skiaVisual = new SKCanvasVisual(this, Visual.Compositor);
+		ElementCompositionPreview.SetElementChildVisual(this, _skiaVisual!);
+	}
+
+	public static DependencyProperty MirroredWhenRightToLeftProperty { get; } = DependencyProperty.Register(
+		nameof(MirroredWhenRightToLeft),
+		typeof(bool),
+		typeof(SKCanvasElement),
+		new FrameworkPropertyMetadata((dO, _) => ((SKCanvasElement)dO).RespectFlowDirectionChanged()));
+
+	/// <summary>
+	/// By default, SKCanvasElement will have the origin at the top-left of the drawing area with the normal directions increasing down and right.
+	/// If MirroredWhenRightToLeft is true, the drawing will be horizontally reflected when <see cref="SKCanvasElement.FlowDirection"/> is <see cref="FlowDirection.RightToLeft"/>.
+	/// </summary>
+	public bool MirroredWhenRightToLeft
+	{
+		get => (bool)GetValue(MirroredWhenRightToLeftProperty);
+		set => SetValue(MirroredWhenRightToLeftProperty, value);
+	}
+
+	private void RespectFlowDirectionChanged()
+	{
+		if (ApplyFlowDirection())
+		{
+			_skiaVisual.Invalidate();
+		}
+	}
+
+	/// <summary>
+	/// The SkiaSharp drawing logic goes here.
+	/// </summary>
+	/// <param name="canvas">The SKCanvas that should be drawn on. The drawing will directly appear in the clipping area.</param>
+	/// <param name="area">The dimensions of the clipping area.</param>
+	protected abstract void RenderOverride(SKCanvas canvas, Size area);
+
+	/// <summary>
+	/// By default, SKCanvasElement uses all the <see cref="availableSize"/> given. Subclasses of SKCanvasElement
+	/// should override this method if they need something different.
+	/// </summary>
+	protected override Size MeasureOverride(Size availableSize)
+	{
+		if (availableSize.Width == Double.PositiveInfinity ||
+			availableSize.Height == Double.PositiveInfinity ||
+			availableSize.Width.IsNaN() ||
+			availableSize.Height.IsNaN())
+		{
+			throw new ArgumentException($"{nameof(SKCanvasElement)} cannot be measured with infinite or NaN values, but received availableSize={availableSize}.");
+		}
+		return availableSize;
+	}
+
+	protected override Size ArrangeOverride(Size finalSize)
+	{
+		_skiaVisual.Size = new Vector2((float)finalSize.Width, (float)finalSize.Height);
+		_skiaVisual.Clip = _skiaVisual.Compositor.CreateRectangleClip(0, 0, (float)finalSize.Width, (float)finalSize.Height);
+
+		ApplyFlowDirection(); // if FlowDirection Changes, it will cause an InvalidateArrange, so we recalculate the TransformMatrix here
+
+		if (finalSize.Width == Double.PositiveInfinity ||
+			finalSize.Height == Double.PositiveInfinity ||
+			finalSize.Width.IsNaN() ||
+			finalSize.Height.IsNaN())
+		{
+			throw new ArgumentException($"{nameof(SKCanvasElement)} cannot be arranged with infinite or NaN values, but received finalSize={finalSize}.");
+		}
+		return finalSize;
+	}
+
+	private bool ApplyFlowDirection()
+	{
+		var oldMatrix = _skiaVisual.TransformMatrix;
+		if (FlowDirection == FlowDirection.RightToLeft && !MirroredWhenRightToLeft)
```

Comment:
Normally, mirroring happen when FlowDirection is the opposite of the parent, which is something we're supposed to already do.

Reply:
It wasn't working when I first added it, but now it's actually working right now without `MirroredWhenRightToLeft`. I will remove it then.

Reply:
Yes, this is supposed to disable this behaviour, since most likely, the user will want to keep the drawing the same regardless of RTL (like an image).